x-pack/legacy/plugins/epm/common/types.ts

```diff
@@ -126,8 +127,8 @@ export interface InstallationAttributes extends SavedObjectAttributes {
   installed: AssetReference[];
 }
 
-export type Datasource = SavedObject<DatasourceAttributes>;
-export interface DatasourceAttributes extends SavedObjectAttributes {
+export type DatasourceSavedObject = SavedObject<DatasourceAttributes>;
```

Comment:
This change causes the following issues:

```
      x-pack/legacy/plugins/epm/common/types.ts:131:18 - error TS2411: Property 'package' of type 'Package' is not assignable to string index type 'SavedObjectAttribute'.

      131 export interface DatasourceAttributes extends SavedObjectAttributes, Datasource {
                           ~~~~~~~~~~~~~~~~~~~~

      x-pack/legacy/plugins/epm/common/types.ts:131:18 - error TS2411: Property 'streams' of type 'Stream[]' is not assignable to string index type 'SavedObjectAttribute'.

      131 export interface DatasourceAttributes extends SavedObjectAttributes, Datasource {
                           ~~~~~~~~~~~~~~~~~~~~

      x-pack/legacy/plugins/epm/server/datasources/create.ts:52:5 - error TS2345: Argument of type '{ installed: Pick<SavedObjectReference, "type" | "id">[]; }' is not assignable to parameter of type 'DatasourceAttributes'.
        Type '{ installed: Pick<SavedObjectReference, "type" | "id">[]; }' is missing the following properties from type 'DatasourceAttributes': id, name, package, streams

      52     { installed: toInstall },
```

Any hints on how to resolve this best? What I would like to have is that we can use our Datasource interface inside  the saved object attributes and just pass it to it.